### PR TITLE
Add mkdocs configuration

### PR DESCRIPTION
**Pull request by jirikuncar**
## Installation
`pip install mkdocs`
## Serve
`mkdocs serve`

### CODE REVIEW

**Comment by nmuesch** [comment]
Worked with @ofek to do similar to integrations-core, where there is a specific theme used for the docs. 
Can now be invoked via tox: `tox -e docs <mkdocs cmd>`

Github action added to build to a gh-pages branch so that its picked up by github pages and can be served via datadoghq.dev/datadog-api-client-python

**Review by jirikuncar — COMMENTED** [review]
👍

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# datadog-api-client-python
+
+Collection of all Datadog Public endpoints.
```
Comment (apigirl) [inline docs/README.md:3]:
```suggestion
Collection of all of the Datadog public endpoints. For the official documentation, see the [Datadog API Reference](https://docs.datadoghq.com/api/).
```

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+This Python package is automatically generated by the [OpenAPI Generator](https://openapi-generator.tech) project:
```
Comment (apigirl) [inline docs/README.md:5]:
```suggestion
This python package is automatically generated by the [OpenAPI Generator](https://openapi-generator.tech) project.
```

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+For more information, please visit [https://www.datadoghq.com/support/](https://www.datadoghq.com/support/)
```
Comment (apigirl) [inline docs/README.md:7]:
```suggestion
Need help? Reach out to our awesome [support team](https://www.datadoghq.com/support/).
```

**Review by apigirl — APPROVED** [review]

**Pull request merged**